pkg/deployments/tasks/20220823-arbitrum-root-gauge-factory-v2/test/task.fork.ts

```diff
@@ -0,0 +1,259 @@
+  });
+
+  it('add gauge to gauge controller', async () => {
+    await gaugeAdder.connect(admin).addGaugeFactory(factory.address, 4); // Arbitrum is Gauge Type 4
+    await gaugeAdder.connect(admin).addArbitrumGauge(gauge.address);
+
+    expect(await gaugeController.gauge_exists(gauge.address)).to.be.true;
+  });
+
+  it('vote for gauge', async () => {
+    expect(await gaugeController.get_gauge_weight(gauge.address)).to.equal(0);
+    expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(0);
+
+    await gaugeController.connect(veBALHolder).vote_for_gauge_weights(gauge.address, 10000); // Max voting power is 10k points
+
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+    // Gauge weight is equal to the cap, and controller weight for the gauge is greater than the cap.
+    expect(
+      await gaugeController['gauge_relative_weight(address,uint256)'](gauge.address, await currentWeekTimestamp())
+    ).to.be.gt(weightCap);
+    expect(await gauge.getCappedRelativeWeight(await currentTimestamp())).to.equal(weightCap);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+
+    const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+    // Even though the gauge has relative weight, it cannot mint yet as it needs for the epoch to finish
+    const bridgeETH = await gauge.getTotalBridgeCost();
+    const zeroMintTx = await authorizerAdaptor
+      .connect(admin)
+      .performAction(gauge.address, calldata, { value: bridgeETH });
+    expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+    await advanceTime(WEEK);
+
+    // The gauge should now mint and send all minted tokens to the Arbitrum bridge
+    const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata, { value: bridgeETH });
+    const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp,
+    });
+    const actualEmissions = event.args.periodEmissions;
+
+    // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+    const expectedEmissions = weightCap.mul(weeklyRate).div(FP_SCALING_FACTOR);
```

Comment:
```suggestion
    // Note that instead of the weight, we use the cap (since we expect for the weight to be larger than the cap)
    const expectedEmissions = weightCap.mul(weeklyRate).div(FP_SCALING_FACTOR);
```